Updated the docs to cover KeyCloak as an OIDC provider (#92606)
openshift/openshift-docs: modules/rosa-hcp-sts-example-external-auth-provider.adoc

// Module included in the following assemblies:
//
// * rosa_hcp/rosa-hcp-sts-creating-a-cluster-quickly.adoc
:_mod-docs-content-type: PROCEDURE
[id="rosa-hcp-sts-example-external-auth-provider_{context}"]
= Example external authentication providers
:source-highlighter: pygments
:pygments-style: emacs
:icons: font

> The following are examples of external authentication provider configurations.

Reviewer: ```suggestion
You can use one of the following examples of external authentication provider configurations to set up your own configuration.

Examples are provided for the following software:

* Microsoft Entra ID
* Keycloak
```
You don't have to use this exactly, but I like the idea of giving the readers their options at the beginning. That way they can scroll to the one they want.

.Example Extra ID configuration

> You can use Microsoft Entra ID as an external provider. You must have already configured an Entra ID server before using it as an external provider. See the link:https://learn.microsoft.com/en-us/entra/identity/?culture=en-us&country=us[Microsoft Entra ID documentation] for more information.

Reviewer: ```suggestion
You can use Microsoft Entra ID as an external provider. You must have already configured a Microsoft Entra ID server before using it as an external provider. See the link:https://learn.microsoft.com/en-us/entra/identity/?culture=en-us&country=us[Microsoft Entra ID documentation] for more information.
```

The following example shows a configured Microsoft Entra ID external authentication provider:

> . Running the following command creates an external authentication provider that uses Keycloak:

Reviewer: ```suggestion
. Create an external authentication provider that uses Keycloak by running the following command:
```

+
[NOTE]
====
You must set your own environment variables with values specific to your Microsoft Entra ID server.
====

> +

Reviewer: .Example input

[source,terminal]
----
$ rosa create external-auth-provider -c $CLUSTER_NAME \  
    --claim-mapping-groups-claim groups \
    --claim-mapping-username-claim <authorized_user_name> \ 
    --console-client-id $CONSOLE_CLIENT_ID \ 
    --console-client-secret $CONSOLE_CLIENT_SECRET_VALUE \ 
    --issuer-audiences "$AUDIENCE_1" \
    --issuer-ca-file ca-bundle.crt --issuer-url $ISSUER_URL \
    --name m-entra-id
----
+

> You should see the following in response:

Reviewer: ```suggestion
.Example output
```

+
[source,terminal]
----
I: Successfully created an external authentication provider for cluster 'ext-auth-test'. It can take a few minutes for the creation of an external authentication provider to become fully effective.
----

> . You can now list the external authentication provider for your cluster to see the issuer URL, or you can use the `rosa describe` command to see all details related to this external authentication provider:

Reviewer: ```suggestion
. List the external authentication provider for your cluster to see the issuer URL or use the `rosa describe` command to see all details related to this external authentication provider by running one of the following commands:
```

> .. List the external authentication configuration on a specified cluster with the following command:

Reviewer: ```suggestion
** List the external authentication configuration on a specified cluster by running the following command:
```

> +

Reviewer: I think the actual command is missing here. Also add `.Example input` before it.

.Example output
[source,terminal]
----
NAME        ISSUER URL
m-entra-id  https://login.microsoftonline.com/<group_id>/v2.0
----

.. Display the external authentication configuration on a specified cluster by using the following command:

> +

Reviewer: .Example input

[source,terminal]
----
$ rosa describe external-auth-provider \
    -c <cluster_name> --name <name_of_external_authentication>
----
+
.Example output
+
[source,terminal]
----
ID:                          ms-entra-id
Cluster ID:                  <cluster_id>
Issuer audiences:
                             - <audience_id>
Issuer Url:                  https://login.microsoftonline.com/<group_id>/v2.0
Claim mappings group:        groups
Claim mappings username:     email
----

.Example Keycloak configuration

You can use Keycloak as an external provider. You must have already configured a Keycloak server before using it as an external provider. See the link:https://www.keycloak.org/server/configuration[Keycloak documentation] for more information.

. Running the following command creates an external authentication provider that uses Keycloak:
+
[NOTE]
====
You must set your own environment variables with values specific to your Keycloak server.
====

> +

Reviewer: .Example input

--claim-mapping-groups-claim groups \ 
    --claim-mapping-username-claim <authorized_user_name> \
    --console-client-id $CONSOLE_CLIENT_ID \ 
    --console-client-secret $CONSOLE_CLIENT_SECRET_VALUE \
    --issuer-audiences "$AUDIENCE_1,$AUDIENCE_2" \ 
    --issuer-ca-file ca-bundle.crt --issuer-url $ISSUER_URL --name keycloak
----
+
You should see the following in response:
+
[source,terminal]
----
I: Successfully created an external authentication provider for cluster 'ext-auth-test'. It can take a few minutes for the creation of an external authentication provider to become fully effective.
----

. You can now list the external authentication provider for your cluster to see the issuer URL, or you can use the `rosa describe` command to see all details related to this external authentication provider:
.. List the external authentication configuration on a specified cluster with the following command:
+
[source,terminal]
----
$ rosa list external-auth-provider -c <cluster_name>
----
+
.Example output
[source,terminal]
----
NAME      ISSUER URL
keycloak  https://keycloak-keycloak.apps.<keycloak_id>.openshift.org/realms/master
----

> .. Display the external authentication configuration on a specified cluster by using the following command:

Reviewer: ```suggestion
** Display the external authentication configuration on a specified cluster by running the following command:
```

> +

Reviewer: .Example input
